File: c/chapter 5/recursion.cpp
//sum of digits using recursion
#include<conio.h>
#include<stdio.h>
int main()
{
   int sumdigit(int x);
   int n,sum;
 
   printf ("Enter five digit Integer number =");
   scanf("%d",&n);
   sum=sumdigit(n);
   printf ("Sum of Digits= %d", sum);
   getch();
}
int sumdigit(int x)
{
  int s=0,d;
  if(x==0)
  { 
   return (0);
  }
  d= x%10;
  s=s+d+sumdigit(x/10);
  return(s);
}

File: java/recursion.java

import java.util.Scanner;
public class recursion{
public static void main(String args[])
{
int num,sum;
System.out.println("enter the 5 digit number");
Scanner in = new Scanner(System.in);
 
num= in.nextInt();

sum=le(num);
System.out.println("sum of 5 digit number" +sum);
}
public static  int le(int a)
{
 int s=0,d;
    if(a==0)
  { 
   return (0);
  }
  d= a%10;
  s=s+d+le(a/10);
  return(s);
}
}